Backpack/Icon/Classes/BPKIcon.m

```diff
@@ -0,0 +1,236 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKIcon.h"
+
+#import <CoreText/CoreText.h>
+
+NS_ASSUME_NONNULL_BEGIN
+NSString * const BPKIconFallbackGlyph = @"\u25A1"; // White box glyph
+
+@interface BPKIcon()
+@property(class, nonatomic, readonly) UIFont *iconFont;
+@property(class, nonatomic, readonly) NSCache<NSString *, UIImage *> *imageCache;
+@property(class, nonatomic, readonly) NSParagraphStyle *paragraphStyle;
+@property(class, nonatomic, readonly, nullable) NSDictionary *iconMapping;
+
++ (NSBundle *)iconBundle;
++ (NSString *)cacheKeyForIconNamed:(NSString *)name withColor:(UIColor *)color size:(BPKIconSize)size;
++ (NSString *)stringForUnicodeCodepoint:(nullable NSString *)codepoint;
+@end
+
+@implementation BPKIcon
+@dynamic imageCache;
+@dynamic iconFont;
+@dynamic paragraphStyle;
+@dynamic iconMapping;
+
++ (void)load {
+    NSBundle *bundle = [self iconBundle];
+
+    NSURL *url = [bundle URLForResource:@"BpkIcon" withExtension:@"ttf"];
+    NSAssert(url != nil, @"`BpkIcon.ttf` must be present in the bundle.");
+    if (url == nil) {
+        return;
+    }
+
+    NSData *fontData = [NSData dataWithContentsOfURL:url];
+
+    CGDataProviderRef provider = CGDataProviderCreateWithCFData((CFDataRef)fontData);
+    CGFontRef font = CGFontCreateWithDataProvider(provider);
+
+    if (font) {
+        CFErrorRef error = NULL;
+        if (CTFontManagerRegisterGraphicsFont(font, &error) == NO) {
+            CFStringRef errorDescription = CFErrorCopyDescription(error);
+            @throw [NSException exceptionWithName:NSInternalInconsistencyException reason:(__bridge NSString *)errorDescription userInfo:@{ NSUnderlyingErrorKey: (__bridge NSError *)error }];
+        }
+
+        CFRelease(font);
+    }
+
+    CFRelease(provider);
+}
+
++ (UIImage *)iconNamed:(NSString *)name color:(UIColor *)color size:(BPKIconSize)size {
+    NSAssert([NSThread isMainThread], @"iconNamed:color:size: should only be called on the main thread.");
```

Comment:
Is there a legitimate reason to prevent loading images on a background thread? 

- `NSCache` is thread-safe
- `UIGraphicsContext` is thread-safe (*"This function may be called from any thread of your app."* - https://developer.apple.com/documentation/uikit/1623912-uigraphicsbeginimagecontextwitho?language=objc)